### PR TITLE
Add SIG network 2021 annual report

### PR DESCRIPTION
**Pull request by caseydavenport**
Fixes https://github.com/kubernetes/community/issues/5506

This PR is currently a template with no questions yet answered, intended to
be a forum for open discussion before codifying the results.

### CODE REVIEW

File: sig-network/annual-report-2021.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+This report reflects back on CY 2020 and was written in April 2021.
+
+> Note: Each response should have supporting documentation, linked KEPs, graphs, data
+> Note: Err on the side of being descriptive with your responses. End users and others that read this might not know our upstream acronyms, shorthand, or where to find a reference. 
+
+# Operational
+
+- How are you doing with operational tasks in [sig-governance.md](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md)?
+  - Is your [README](./README.md) accurate?
```
Comment (caseydavenport) [inline sig-network/annual-report-2021.md:11]: Yes, README looks accurate.

File: sig-network/annual-report-2021.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+This report reflects back on CY 2020 and was written in April 2021.
+
+> Note: Each response should have supporting documentation, linked KEPs, graphs, data
+> Note: Err on the side of being descriptive with your responses. End users and others that read this might not know our upstream acronyms, shorthand, or where to find a reference. 
+
+# Operational
+
+- How are you doing with operational tasks in [sig-governance.md](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md)?
+  - Is your [README](./README.md) accurate? 
+  - Have a CONTRIBUTING.md file?
```
Comment (caseydavenport) [inline sig-network/annual-report-2021.md:12]: No CONTRIBUTING.md file - we need to add this.

File: sig-network/annual-report-2021.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  - All subprojects correctly mapped and listed in [sigs.yaml](https://github.com/kubernetes/community/blob/master/sig-list.md)?
```
Comment (caseydavenport) [inline sig-network/annual-report-2021.md:13]: Missing the network plumbing WG - @dcbw is this still active?

File: sig-network/annual-report-2021.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  - What’s your meeting culture? Large/small, active/quiet, learnings? Meeting notes up to date? Are you keeping recordings up to date/trends in community members watching recordings?
```
Comment (caseydavenport) [inline sig-network/annual-report-2021.md:14]: Fairly large meetings, with a handful of regular speakers. Pretty active. 

We have an agenda doc, but written minutes are few-and-far-between. Could probably do with better note taking. 

Recordings are up-to-date and posted automatically.

File: sig-network/annual-report-2021.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+- How does the group get updates, reports, or feedback from subprojects? Are there any springing up or being retired? Are OWNERS files up to date in these areas?
```
Comment (caseydavenport) [inline sig-network/annual-report-2021.md:16]: We get periodic updates in the bi-weekly "all-hands" meeting, but they are push not pull (i.e., up to the subproject to bring items to the broader group's attention). No formal reporting procedure. 

@dcbw @jayunit100 @bowei could you guys comment on OWNER files for your respective sub-groups?

File: sig-network/annual-report-2021.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+- Same question as above but for working groups.
+- When was your last monthly community-wide update? (provide link to deck and/or recording)
```
Comment (caseydavenport) [inline sig-network/annual-report-2021.md:18]: oof.

File: sig-network/annual-report-2021.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+# Membership
+
+- Are all listed SIG leaders (chairs, tech leads, and subproject owners) active?
+- How do you measure membership? By mailing list members, OWNERs, or something else?
```
Comment (caseydavenport) [inline sig-network/annual-report-2021.md:23]: I am not aware of us measuring or tracking membership in any explicit way.

File: sig-network/annual-report-2021.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+- How does the group measure reviewer and approver bandwidth? Do you need help in any area now? What are you doing about it?
+- Is there a healthy onboarding and growth path for contributors in your SIG? What are some activities that the group does to encourage this? What programs are you participating in to grow contributors throughout the contributor ladder?
```
Comment (caseydavenport) [inline sig-network/annual-report-2021.md:25]: Regular triage is probably the most we do at the moment - basically new members can volunteer to help walk a user through gathering information about a potential bug / enhancement and thus classify the issue. 

So, not a ton.

File: sig-network/annual-report-2021.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+- What programs do you participate in for new contributors?
+- Does the group have contributors from multiple companies/affiliations? Can end users/companies contribute in some way that they currently are not?
```
Comment (caseydavenport) [inline sig-network/annual-report-2021.md:27]: Yes - we definitely have a diverse group.

**Review by caseydavenport — COMMENTED** [review]
A very very quick first-pass to hopefully get the conversation started.

File: sig-network/annual-report-2021.md (hunk 5 above)
Comment (robscott) [reply sig-network/annual-report-2021.md:16]: Gateway API OWNERS file is up to date. Also happy to give more regular/formal updates for Gateway API progress.

File: sig-network/annual-report-2021.md (hunk 3 above)
Comment (dcbw) [reply sig-network/annual-report-2021.md:13]: @caseydavenport It's not technically an appproved Kube working group... we just started using the same Zoom 3+ years ago and never moved elsewhere. We can discuss starting that process though?

File: sig-network/annual-report-2021.md (hunk 4 above)
Comment (dcbw) [reply sig-network/annual-report-2021.md:14]: I usually try to take notes on action items and important points, but we can always do better.

File: sig-network/annual-report-2021.md (hunk 6 above)
Comment (dcbw) [reply sig-network/annual-report-2021.md:18]: Yeah, we fell out of this one.

Comment (bridgetkromhout) [reply sig-network/annual-report-2021.md:18]: Do we want to schedule one and then say "we've not had one in a while but will do one at X date"?

File: sig-network/annual-report-2021.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+# Current initiatives and project health
+
+> Please include links to KEPs and other supporting information that will be beneficial to multiple types of community members. 
+
+- What are initiatives that should be highlighted, lauded, shout out, that your group is proud of? Currently underway? What are some of the longer tail projects that your group is working on?
+- Year to date KEP work review: What’s now stable? Beta? Alpha? Road to alpha?
+- What areas and/or subprojects does the group need the most help with?
```
Comment (dcbw) [inline sig-network/annual-report-2021.md:35]: PR reviews; metrics on issues/PRs; reporting to other SIGs...

File: sig-network/annual-report-2021.md (hunk 6 above)
Comment (bridgetkromhout) [reply sig-network/annual-report-2021.md:18]: As I said on the call, I'm happy to help with this - putting the presentation/session together.

File: sig-network/annual-report-2021.md (hunk 8 above)
Comment (bridgetkromhout) [reply sig-network/annual-report-2021.md:25]: Over the summer we encouraged new members by helping beginners run triage in the meetings.

File: sig-network/annual-report-2021.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+- Is there a healthy onboarding and growth path for contributors in your SIG? What are some activities that the group does to encourage this? What programs are you participating in to grow contributors throughout the contributor ladder?
+- What programs do you participate in for new contributors?
+- Does the group have contributors from multiple companies/affiliations? Can end users/companies contribute in some way that they currently are not?
+
+# Current initiatives and project health
+
+> Please include links to KEPs and other supporting information that will be beneficial to multiple types of community members. 
+
+- What are initiatives that should be highlighted, lauded, shout out, that your group is proud of? Currently underway? What are some of the longer tail projects that your group is working on?
+- Year to date KEP work review: What’s now stable? Beta? Alpha? Road to alpha?
```
Comment (bridgetkromhout) [inline sig-network/annual-report-2021.md:34]: Dual-stack is beta and headed towards stable later this year.

File: sig-network/annual-report-2021.md (hunk 4 above)
Comment (bridgetkromhout) [reply sig-network/annual-report-2021.md:14]: Do we want to supplement the notes with automated transcripts out of zoom? https://support.zoom.us/hc/en-us/articles/115004794983-Using-audio-transcription-for-cloud-recordings-

**Comment by bridgetkromhout** [comment]
@caseydavenport I'm offline tomorrow but happy to take a look and comment/add things Monday if it's still in flight.

File: sig-network/annual-report-2021.md (hunk 1 above)
Comment (jayunit100) [reply sig-network/annual-report-2021.md:11]: copying here for collab https://hackmd.io/SITAZvJ8RLWJy1K5FFmiIQ

Comment (caseydavenport) [reply sig-network/annual-report-2021.md:11]: Let's please keep comments on the PR. I don't want to have to collect comments from a bunch of different locations. I also don't want to have to make a new account to use that site.

File: sig-network/annual-report-2021.md (hunk 3 above)
Comment (caseydavenport) [reply sig-network/annual-report-2021.md:13]: @dcbw probably makes sense to make it official :grin:

**Comment by caseydavenport** [comment]
@kubernetes/sig-network-misc @bridgetkromhout @dcbw @thockin @robscott I've updated the PR based on the first round of comments and discussion and whatever I could fill in myself. Please take a look and add some comments if you have any!

File: sig-network/annual-report-2021.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+- Does the group have contributors from multiple companies/affiliations? Can end users/companies contribute in some way that they currently are not?
+  - Yes - we have long-term as well as one-time contributors from a variety of companies. 
+
+# Current initiatives and project health
+
+> Please include links to KEPs and other supporting information that will be beneficial to multiple types of community members. 
+
+- What are initiatives that should be highlighted, lauded, shout out, that your group is proud of? Currently underway? What are some of the longer tail projects that your group is working on?
+- Year to date KEP work review: What’s now stable? Beta? Alpha? Road to alpha?
+  - EndpointSlice: [GA](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/0752-endpointslices)
```
Comment (bowei) [inline sig-network/annual-report-2021.md:53]: gateway reached v1alpha1

cc" @robscott

Comment (robscott) [reply sig-network/annual-report-2021.md:53]: Good call, TopologyAwareHints also reached alpha

Comment (robscott) [reply sig-network/annual-report-2021.md:53]: If this is covering 2020, we should also probably mention that Ingress reached GA.

File: sig-network/annual-report-2021.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+  - Multiple CIDRs: [road to alpha](https://github.com/kubernetes/enhancements/pull/2594)
+  - All-port services: [road to alpha](https://github.com/kubernetes/enhancements/pull/2611/)
+  - Topology-aware-hints: [Alpha](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2433-topology-aware-hints)
+  - Ingress: [GA](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1453-ingress-api)
+- What areas and/or subprojects does the group need the most help with?
+  - Tracking project health, membership, progress.
+  - PR reviews
+  - Reporting to other SIGs / community meeting.
+- What's the average open days of a PR and Issue in your group? / what metrics does your group care about and/or measure?
+  - I don't think we currently track this.
```
Comment (mrbobbytables) [inline sig-network/annual-report-2021.md:65]: Not necessarily something you need to add, but a few useful dashboards:
- [PR Workloads per SIG](https://k8s.devstats.cncf.io/d/34/pr-workload-per-sig-table?orgId=1&var-period_name=Last%20year)
- [PR Time to Approve](https://k8s.devstats.cncf.io/d/44/pr-time-to-approve-and-merge?var-period=w&var-repogroup_name=SIG%20Network&var-repo_name=kubernetes%2Fkubernetes&var-apichange=All&var-size_name=All&var-kind_name=All) (This is not very accurate, but can be an indicator)

**Review by mrbobbytables — COMMENTED** [review]
Thanks!, I have a couple non-blocking suggestions / comments :) More so FYIs

File: sig-network/annual-report-2021.md [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+# Operational
+
+- How are you doing with operational tasks in [sig-governance.md](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md)?
+  - Is your [README](./README.md) accurate? 
+    - Yes, our README is up to date.
+  - Have a CONTRIBUTING.md file?
+    - No - we do not have a CONTRIBUTING.md file.
+  - All subprojects correctly mapped and listed in [sigs.yaml](https://github.com/kubernetes/community/blob/master/sig-list.md)?
+    - Yes, all subprojects are correctly listed. 
+    - We have one unofficial group that should probably be a subproject - the network plumbing group.
```
Comment (derekwaynecarr) [inline sig-network/annual-report-2021.md:17]: Can you provide a link to any materials?
Are meetings open and available for all to attend?
Can the SIG take an action to clarify its future?

Comment (derekwaynecarr) [reply sig-network/annual-report-2021.md:17]: Are meetings recorded and uploaded similar to normal SIG meetings?

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
please link to any pertinent materials for network plumbing wg.

it would be good if we can clarify its official/unofficial status as an action out of this annual report.

File: sig-network/annual-report-2021.md (hunk 14 above)
Comment (knobunc) [reply sig-network/annual-report-2021.md:17]: @dcbw @dougbtv @squeed -- Do you have an answers to Derek's questions?

The agenda and minutes are at: https://docs.google.com/document/d/1oE93V3SgOGWJ4O1zeD1UmpeToa0ZiiO6LqRAmZBPFWM/edit and that has the meeting details.

File: sig-network/annual-report-2021.md [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+    - Recordings are automatically uploaded to the YouTube channel and are up to date.
+    - Notes are stored in the minutes document, which is also where topics can be added to the agenda in advance.
+
+- How does the group get updates, reports, or feedback from subprojects? Are there any springing up or being retired? Are OWNERS files up to date in these areas?
+  - Updates from subprojects are generally the responsibilitiy of subproject members themselves (i.e., push not pull).
+  - Updates are usually on the mailing list, occasionally subprojects will also use meeting time to discuss in real-time.
+- Same question as above but for working groups.
+  - Same answer as above, but for working groups.
+- When was your last monthly community-wide update? (provide link to deck and/or recording)
+  - Last community meeting was long enough ago that it is no longer in the minutes history (so, clearly we're due).
```
[the remark this answers is not in the document]
Comment (caseydavenport) [reply sig-network/annual-report-2021.md:30]: Thanks for tracking this down - will add to the report.

**Comment by derekwaynecarr** [comment]
thanks for updates and related links.

/approve

**Pull request merged**